contracts/StakerV2.sol

```diff
@@ -0,0 +1,948 @@
+    @param amount amount of the pool asset being provided by the user.
+    @param tokenBoostedAmount tokens amount after boosts.
+    @param pointBoostedAmount points amount after boosts.
+    @param tokenPaid value of user's total token earnings paid out. 
+    pending reward = (user.amount * pool.tokensPerShare) - user.rewardDebt.
+    @param pointPaid value of user's total point earnings paid out.
+  */
+  struct UserInfo {
+    uint256 amount;
+    uint256 tokenBoostedAmount;
+    uint256 pointBoostedAmount;
+    uint256 tokenPaid;
+    uint256 pointPaid;
+  }
+
+  // Stored information for each user staking in each pool.
+  mapping (IERC20 => mapping (address => UserInfo)) public userInfo;
+
+  // The total sum of the strength of all pools.
+  uint256 public totalTokenStrength;
+  uint256 public totalPointStrength;
+
+  // The total amount of the disbursed token ever emitted by this Staker.
+  uint256 public totalTokenDisbursed;
+
+  // Users additionally accrue non-token points for participating via staking.
+  mapping (address => uint256) public userPoints;
+  mapping (address => uint256) public userSpentPoints;
+
+  // A map of all external addresses that are permitted to spend user points.
+  mapping (address => bool) public approvedPointSpenders;
+
+  /**
+    The type of asset on which the boost is applied.
+    @param Tokens boost is applied on the disburse token.
+    @param Points boost is applied on the points.
+    @param Both boost is applied on both.
+  */
+  enum AssetType {
+    Tokens,
+    Points,
+    Both
+  }
+
+  /** 
+    A booster struct, which stores information on the boost requirements.
+    @param set if set, then the booster is active.
+    @param multiplier the rate which will act as a boost on basis points.
+    A multiplier of Zero means, the booster is not set.
+    @param amountRequired number of NFTs required from the contract.
+    @param groupRequired (optional) specifies a group from NFT contract 
+      as requirement for the boost. If 0, then any group or item.
+    @param contractRequired contract that the required assets belong to.
+    @param assetType enum that specifies Tokens/Points to boost or both.
+  */
+  struct BoostInfo {
+    uint256 multiplier;
+    uint256 amountRequired;
+    uint256 groupRequired;
+    address contractRequired;
+    AssetType assetType;
+  }
+
+  // Mapping of Booster ID to its 'BoostInfo'.
+  mapping (uint256 => BoostInfo) public boostInfo;
+
+  // Number of boosters that are active.
+  uint256 public activeBoosters;
+
+  /**
+    A struct containing the NFT staker information.
+    @param boosterIds enumerable booster Ids the user has staked NFTs in.
+    @param tokenIds actual ids of tokens mapped to booster Id.
+    @param totalNfts total NFTs staked in this contract by NFT staker.
+    @notice contract information at the time of unstaking is retreived
+    from the boostInfo.
+   */
+  struct NftUserInfo {
```

Comment:
In general Yes. Solidity tries to pack up variables inside a struct. For example: A bool followed by a uint256 will cost 2 slots. A bool followed by uint128 will cost 1 slot. For NftUserInfo, I need to change it a little bit. 

It works like a tetris game. If two variables can fit in a 32Byte slot, gas is saved since each storage slot costs gas.